### PR TITLE
fix: standardized capitalization issues with config file

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
fixes an issue when the config file and other properties passed in weren't matching due to a capitalization difference
### What issues does this PR fix or reference?
@W-8659187@

### CODE REVIEW

**Review by mdonnalley — APPROVED** [review]

File: src/commands/force/user/create.ts [hunk 1]
```diff
@@ -151,19 +151,24 @@ export class UserCreateCommand extends SfdxCommand {
...
+  }
+
   private async aggregateFields(defaultFields: UserFields): Promise<UserFields & Dictionary<string>> {
     // start with the default fields, then add the fields from the file, then (possibly overwritting) add the fields from the cli varargs param
     if (this.flags.definitionfile) {
       const content = await fs.readJson(this.flags.definitionfile);
       Object.keys(content).forEach((key) => {
-        defaultFields[key] = content[key];
+        // cast entries to lowercase to standardize
+        defaultFields[this.lowerFirstLetter(key)] = content[key];
```
Comment (amphro) [inline src/commands/force/user/create.ts:164]: The API is case insensitive too right?

**Review by amphro — APPROVED** [review]

File: src/commands/force/user/create.ts [hunk 2]
```diff
@@ -123,9 +123,9 @@ export class UserCreateCommand extends SfdxCommand {
     }
 
     // Set the alias if specified
-    if (this.flags.setalias) {
+    if (this.flags.setalias || fields.alias) {
```
Comment (shetzel) [inline src/commands/force/user/create.ts:126]: `fields.alias` is the alias in the User metadata api, not the alias that the cli uses.  They are 2 different things.

File: test/commands/user/create.test.ts [hunk 3]
```diff
@@ -5,13 +5,62 @@
...
 import { $$, expect, test } from '@salesforce/command/lib/test';
 import { Aliases, Connection, DefaultUserFields, fs, Org, User } from '@salesforce/core';
 import { stubMethod } from '@salesforce/ts-sinon';
+import { IConfig } from '@oclif/config';
+import UserCreateCommand from '../../../src/commands/force/user/create';
 
 const username = 'defaultusername@test.com';
 
 describe('force:user:create', () => {
+  it('will properly merge fields regardelss of capitialization', async () => {
```
Comment (shetzel) [inline test/commands/user/create.test.ts:19]: nit: regardelss --> regardless

**Review by shetzel — CHANGES_REQUESTED** [review]

File: src/commands/force/user/create.ts (hunk 2 above)
Comment (WillieRuemmele) [reply src/commands/force/user/create.ts:126]: ahhh, I thought I missed this the first time and was adding it back 👍

**Review by shetzel — APPROVED** [review]

**Pull request merged**